libsql/src/replication/mod.rs

```diff
@@ -256,13 +256,23 @@ impl EmbeddedReplicator {
             }
         }
 
-        let last_frames_synced = self.last_frames_synced.fetch_add(
-            replicator.frames_synced(),
-            std::sync::atomic::Ordering::Relaxed,
-        );
+        let current_frames_synced = replicator.frames_synced();
+
+        let last_frames_synced = self
+            .last_frames_synced
+            .load(std::sync::atomic::Ordering::Relaxed);
+
+        if current_frames_synced > last_frames_synced {
```

Comment:
This needs to be a while loop. You fetch last_frames_synced, compare with current_frames_synced and if it's smaller then compare_exchange and if it succeeds then break the loop.